subprojects/smoke-test/src/smokeTest/resources/org/gradle/smoketests/validate-external-gradle-plugin.gradle.kts

```diff
@@ -67,17 +67,17 @@ fun configureValidationTask(project: Project,
     val idWithoutDots = pluginId.replace('.', '_')
     return project.tasks.register<ValidatePlugins>("validatePluginWithId_" + idWithoutDots) {
         group = "Plugin development"
-        outputFile.set(project.layout.buildDirectory.file("reports/plugins/validation-report-for-$idWithoutDots.txt"))
+        outputFile = project.layout.buildDirectory.file("reports/plugins/validation-report-for-$idWithoutDots.txt")
 
         val scriptHandler = project.buildscript as ScriptHandlerInternal
         val scriptClassPath = scriptHandler.scriptClassPath.asFiles
-        classpath.setFrom(scriptClassPath)
+        classpath = project.files(scriptClassPath)
 
         val archiveOperations = findArchiveOperations(project)
         val pluginClassesOf = pluginJars.stream()
             .map { zipPath: File? -> archiveOperations.zipTree(zipPath!!) }
             .collect(Collectors.toList())
-        classes.setFrom(pluginClassesOf)
+        classes = project.files(pluginClassesOf)
```

Comment:
❓ Why do we need to call `project.files()` here? Can we make it so that `=` works for any value that `setFrom()` accepts?

Reply:
ConfigurableFileCollection.setFrom is pretty unrestrictive:
```
void setFrom(Iterable<?> paths);
void setFrom(Object... paths);
```

To support that case pointed out we would need to define:
```
fun ConfigurableFileCollection.assign(iterable: Iterable<File>)
```
that is maybe something that makes sense.

Reply:
FTR, we discussed that we already have `ConfigurableFileCollection.assign(fc: FileCollection)` supported since 8.1, and that should be good. Meanwhile `pluginClassesOf` is not a `FileCollection`, so calling `files()` here makes sense.